parquet/src/main/java/org/apache/iceberg/data/parquet/BaseParquetReaders.java

```diff
@@ -76,6 +67,31 @@ protected ParquetValueReader<T> createReader(
   protected abstract ParquetValueReader<T> createStructReader(
       List<Type> types, List<ParquetValueReader<?>> fieldReaders, Types.StructType structType);
 
+  protected ParquetValueReaders.PrimitiveReader<?> fixedReader(ColumnDescriptor desc) {
+    return null;
+  }
+
+  protected ParquetValueReaders.PrimitiveReader<?> int96Reader(ColumnDescriptor desc) {
```

Comment:
These should all return `ParquetValueReader<?>` rather than constrain the implementation to a subclass of `PrimitiveReader`.